### ISSUE TITLE
Imported Pretrained Keras Model Produces Wrong Output in DL4J

### ISSUE THREAD

**Issue by j1234d2003** [deeplearning4j/deeplearning4j#8692]
#### Issue Description

Keras prediction outputs don’t match DL4J outputs for a CNN model created and trained in Keras and imported in DL4J.

I have a Keras 2.3.1 h5 CNN fully trained model that I import using DL4J 1.0.0-beta6 version. I intend to use DL4J for evaluating outputs for specific input images. Currently Keras prediction outputs don’t match DL4J outputs. The shapes of the inputs are different Keras input shape is (4, 150, 150, 3) whereas DL4J input shape is [4, 3, 150, 150]. 

I could not find any sample DL4J code which performs predictions on a CNN (or any network) trained model imported from Keras that use images for predictions in Java and hence cannot ascertain if this is an issue. Beyond that, I cannot identify any other reason for this discrepency and I would greatly appreciate any help in this regards.

Questions:

1) What is the right shape of input required ini DL4J for the imported Keras Model that expects image input.

2) If input shapes have to match between DL4J and Keras, I cannot identify a way to build a “channel last” INDArray from a set of input images.

3) I need to also perform this on Keras imported mixed multi-input and multi-output networks. What is recommended way to feed inputs to such networks for predictions.

#### Version Information

Keras 2.3.1
Deeplearning4j 1.0.0-beta6 
Platform: Ubuntu 16.04 and 18.04

#### Additional Information

Please refer to [Gist](https://gist.github.com/j1234d2003/661f953525fcb4e30d01a534f76ec573) at https://gist.github.com/j1234d2003/661f953525fcb4e30d01a534f76ec573 which includes the model and test code and test results that demonstrates this problem.

**Comment by AlexDBlack** [comment]
Looking at the type of layers you have here (which are all well tested), my first guess would be the data pipeline differences - you already noted the NCHW vs. NHWC layout, but another thing to be aware of is RGB vs. BGR for the input images. IIRC NativeImageLoader has methods for both.
Plus differences in normalization of course can cause differences here.

One thing I can suggest is preprocessing some images in Python, saving them to Numpy format, and then loading them with Nd4j.readNumpy before performing inference (instead of using NativeImageLoader).
If the network output is the same using that, then we can be reasonably confident that the difference is simply in the data pipeline (normalization or RGB vs. BGR)

**Comment by treo** [comment]
Answered in https://community.konduit.ai/t/imported-pretrained-keras-model-produces-wrong-output-in-dl4j/120/5

It was because of the RGB vs BGR difference.

Closing here, because the user cross posted.

**Comment by saudet** [comment]
So this is a duplicate of #8258, etc. We can easily solve that this way: https://github.com/eclipse/deeplearning4j/issues/8258#issuecomment-540260738
